### ISSUE TITLE
Add some documentation/Explanation - What is the purpose of the "Helpful" button?

### ISSUE THREAD

**Issue by matthieuHenocque** [continuedev/continue#977]
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

I've been using the Continue extension in VSCode on personal and professional project for a couple of weeks, and I have a few improvements suggestion already, but for now I want to focus on a specific one: Documentation.

I struggle to figure out the purpose of the helpful/unhelpful button. I understand what it does for the time of the session, but I have no idea if I'm teaching the model I use, if I'm teaching the context for the time of the session or if that context is saved and reloaded when I switch from one session to another.

I wish it teaches the model. I hope it is saved and reloaded upon sessions.

What about changing model on the server midway ? I have no clue what I'm doing, what I'm losing and if I should use the feature carefully to teach my model.

Lastly, I wish there were 2 things
- A way to delete/hide parts of the conversation in the session history
- A way to say "That was a great session, let's save this forever in my model's brain so that I can remember it of it in 3 months when I'll be working on the subject again"

Anyway, thank you very very much for your work. This is a game changer for me, and will probably be a game changer for my company too. I wish you the best of luck.

### Solution

Add some documentation somewhere. That feature looks so important and yet there's no information on how it works.

**Comment by dschilling** [comment]
Looks like those buttons send feedback to the Continue developers. See https://github.com/continuedev/continue/blob/addbdcb9f517a1534a7e6dd5ab323011faa5314e/gui/src/components/StepContainer/FeedbackButtons.tsx#L24 . So probably this information is used for improving the Continue extensions themselves - maybe for tweaking the prompts, or for evaluating new models when they come out.

It would be super cool if that feedback was used to fine-tune your model, but there would be a lot of moving parts to get that to work.

First, just a "Helpful/Unhelpful" flag is not enough information to use for fine-tuning. More than knowing just that the output was unhelpful, the model needs to be taught what the output *should have* been. These flags could go into a review queue and be reviewed by human reviewers who would create quality responses to be used when training the model. But humans are expensive, and there could be privacy concerns. Instead of human reviewers, some kind of LLM review workflow could be built - maybe simply asking a smarter model to generate a response from the problematic prompt, or asking an LLM what problems it can identify with the response, and then ask it to address those problems to create a better response to use for training data.

Second, Continue is extremely flexible with regards to what models you use. Anthropic, OpenAI, Mistral, or myriads of different local options via Ollama. I don't have experience with any of these vendors. It does look like some of them provide ways to create fine tuned versions of their models. It's definitely possible with a local model, but that's going to take some compute time, etc.

Finally, Continue's config would need to be pointed at the newly fine-tuned model.

While it would be a super cool feature, it would be pretty complicated to implement. I'm just a random guy, so can't speak for the Continue developers, but I really doubt a feature like that would be added.
